packages/token/src/erc721/extensions/erc721_enumerable/interface.cairo

```diff
@@ -0,0 +1,15 @@
+// SPDX-License-Identifier: MIT
+// OpenZeppelin Contracts for Cairo v0.15.0-rc.0
+// (token/erc721/extensions/erc721_enumerable/interface.cairo)
+
+use starknet::ContractAddress;
+
+pub const IERC721ENUMERABLE_ID: felt252 =
+    0x16bc0f502eeaf65ce0b3acb5eea656e2f26979ce6750e8502a82f377e538c87;
+
+#[starknet::interface]
+pub trait IERC721Enumerable<TState> {
+    fn total_supply(self: @TState) -> u256;
+    fn token_by_index(self: @TState, index: u256) -> u256;
+    fn token_of_owner_by_index(self: @TState, owner: ContractAddress, index: u256) -> u256;
```

Comment:
In the example, you mean 1 `balance_of` call and 100 storage reads vs 100 `balance_of`/storage reads, right? I think it's an interesting idea for optimization. I'd prefer getting the core functionality merged first before considering non-essential additions. WDYT?

Reply:
Actually, I don't see why 100 `balance_of` could be necessary. We assume balance doesn't change while the list of owned token is being built, so we can safely call `balance_of` only once at the beginning

I was talking about an optimisation on the Frontend side. With 100 tokens it's going to take 101 RPC calls and with a helper function it will take only a single one

I agree it doesn't have to be part of this PR thought. WDYT about adding an additional helper read function for building a list of all tokens owned by a user later?

Reply:
> I was talking about an optimisation on the Frontend side... I agree it doesn't have to be part of this PR thought. WDYT about adding an additional helper read function for building a list of all tokens owned by a user later?

For sure! I don't see a strong reason not to, but this should be considered low priority at least for now